### PR TITLE
Additions to docs/README

### PR DESCRIPTION
**Pull request by myNameIsPatrick**
## Description

This PR adds some additions to the documentation, in particular adding an authentication and message formats section. In addition, the README now makes explicit how to use the python API in addition to the command line interface.

Also had to update `requirements.txt` as that is used by readthedocs to generate the API documentation.

## Checklist

* [x] All new functions and classes are documented and adhere to Google doc style (3.8.3-3.8.6 of [this](http://google.github.io/styleguide/pyguide.html#383-functions-and-methods) document)
* [x] Add/update sphinx documentation with any relevant changes.
* [x] Add/update pytest-style tests in `/tests`, ensuring sufficient code coverage.
* [x] `make test` runs without errors.
* [x] `make lint` doesn't give any warnings.
* [x] `make format` doesn't give any code formatting suggestions.
* [x] `make doc` runs without errors and generated docs render correctly.
* [x] Check that CI pipeline run on this PR passes all stages.
* [ ] Review signoff by at least one developer.

NOTE: If this PR relates to a release, open and reference an issue with the Release checklist template.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +40,86 @@ hop subscribe kafka://hostname:port/gcn -s EARLIEST
...
+```python
+    from hop import stream
+
+    with stream.open("kafka://hostname:port/topic", "r") as s:
+        for message in s:
+             print(message)
+```
+
+This will listen to the Hop broker, listening to new messages and printing them to
+stdout as they arrive until there are no more messages in the stream.
```
Comment (shereenElSayed) [inline README.md:108]: I think the stream will still be open even if there are no more messages (open forever), right? if so, I guess we need to rephrase this sentence.

File: README.md [hunk 2]
```diff
@@ -38,6 +40,86 @@ hop subscribe kafka://hostname:port/gcn -s EARLIEST
...
+    from hop import stream
+    from hop.io import StartPosition
+
+    with stream.open("kafka://hostname:port/topic", "r", start_at=StartPosition.EARLIEST) as s:
+        for message in s:
+             print(message)
+```
+
+
 ## Installation
```
Comment (shereenElSayed) [inline README.md:123]: I think we need the installation section first before getting into the details of authorization and APIs.

File: requirements.txt [hunk 3]
```diff
@@ -1 +1,3 @@
-adc==1.0.0
+adc-streaming>=1.0.2
```
Comment (shereenElSayed) [inline requirements.txt:1]: Are we now okay with adc-streaming versions >1.0.0?

File: README.md (hunk 1 above)
Comment (myNameIsPatrick) [reply README.md:108]: No, it'll close now by default when it reads all the messages in the stream thanks to the awesome work Spencer did in adc-streaming 1.x. To listen to messages forever, now we need the `persist=True` flag in the open connection.

Comment (shereenElSayed) [reply README.md:108]: okay, it's clearer now to me. Thanks.

File: requirements.txt (hunk 3 above)
Comment (myNameIsPatrick) [reply requirements.txt:1]: Yeah, I think so. The strict requirements pinning there was added for some reason, which isn't particularly useful here. I think it was generated from doing a `pip freeze` but the package isn't called `adc`, it's `adc-streaming` so the readthedocs bot crashes here.

File: README.md (hunk 2 above)
Comment (myNameIsPatrick) [reply README.md:123]: I've been mostly following how some other projects (e.g. https://github.com/robinhood/faust) structure their sections. Not sure why they put their installation section later, but I'm happy to change it if you prefer it up top.

Comment (shereenElSayed) [reply README.md:123]: I'd prefer it if we move it upward, it makes more sense to me. If you wish, we can get another opinion :)
I looked at https://github.com/fastavro/fastavro for instance or https://github.com/marcosschroh/python-schema-registry-client

File: requirements.txt (hunk 3 above)
Comment (shereenElSayed) [reply requirements.txt:1]: okay cool

File: README.md (hunk 2 above)
Comment (myNameIsPatrick) [reply README.md]: Nah that's fine, after rearranging the sections I think it looks better also. I updated the README accordingly.

**Comment by myNameIsPatrick** [comment]
Thanks @shereenElSayed for the helpful comments and the speedy review!

**Comment by myNameIsPatrick** [comment]
For some reason the codecov check is failing, but it's an issue connecting with codecov so nothing to be worried about. Going to merge this in.

**Pull request merged**